lib/decode/decode.go

```diff
@@ -120,15 +122,39 @@ func HookWeakDecodeFromSlice(from, to reflect.Type, data interface{}) (interface
 
 	switch d := data.(type) {
 	case []map[string]interface{}:
-		switch {
-		case len(d) == 0:
-			return nil, nil
-		case len(d) == 1:
-			return d[0], nil
-		default:
-			return data, nil
+		if len(d) == 1 {
```

Comment:
Is there a semantic difference between returning `nil` or `data` when `len(d) == 0`? Notice that changed in this commit

Reply:
There shouldn't be. I realized the old case for `len(d) == 0` wasn't any different than the default case. Mapstructure will use nil or the empty value the same way, so I removed the extra condition